### PR TITLE
Reuse LocalAlluxioCluster to speedup REST tests

### PR DESCRIPTION
**Pull request by gpang**
This speeds up the REST tests, by reusing the LocalAlluxioCluster throughout each test class. This avoids tearing down and recreating the cluster between each individual test case. Minor modifications were required to be able to re-use an existing LocalAlluxioCluster between tests.

### CODE REVIEW

File: tests/src/test/java/alluxio/client/rest/JobMasterClientRestApiTest.java [hunk 1]
```diff
@@ -136,6 +153,6 @@ private void waitForStatus(final long jobId, final Status status)
         Throwables.propagate(e);
       }
       return null;
-    }, WaitForOptions.defaults().setTimeoutMs(10 * Constants.SECOND_MS));
+    }, WaitForOptions.defaults().setInterval(10).setTimeoutMs(10 * Constants.SECOND_MS));
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/client/rest/JobMasterClientRestApiTest.java]: why is this needed?

File: tests/src/test/java/alluxio/client/rest/JobMasterRestApiTest.java [hunk 2]
```diff
@@ -31,6 +35,15 @@
...
 
+  // TODO(chaomin): Rest API integration tests are only run in NOSASL mode now. Need to
+  // fix the test setup in SIMPLE mode.
+  @Rule
+  public LocalAlluxioClusterResource mResource = new LocalAlluxioClusterResource.Builder()
+      .setProperty(PropertyKey.SECURITY_AUTHORIZATION_PERMISSION_ENABLED, "false")
+      .setProperty(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.NOSASL.getAuthName())
+      .setProperty(PropertyKey.USER_FILE_BUFFER_BYTES, "1KB")
+      .build();
+
```
Comment (ZacBlanco) [inline tests/src/test/java/alluxio/client/rest/JobMasterRestApiTest.java]: Does this test even need an Alluxio cluster?

File: tests/src/test/java/alluxio/client/rest/JobMasterClientRestApiTest.java (hunk 1 above)
Comment (gpang) [reply tests/src/test/java/alluxio/client/rest/JobMasterClientRestApiTest.java]: I thought it made the test faster, but not really. I changed it back.

File: tests/src/test/java/alluxio/client/rest/JobMasterRestApiTest.java (hunk 2 above)
Comment (gpang) [reply tests/src/test/java/alluxio/client/rest/JobMasterRestApiTest.java]: I tried removing it, but the test does not run. I think the cluster sets up all the journaling correctly, which the job cluster does not (yet).

**Comment by ZacBlanco** [comment]
(Please update the PR description before merging)

**Comment by gpang** [comment]
alluxio-bot, merge this please
